### PR TITLE
Chore/code review

### PR DESCRIPTION
**Pull request by nopcoder**

### CODE REVIEW

File: batchread/batchread.go [hunk 1]
```diff
@@ -112,8 +115,11 @@ func readEntriesBatch(inputBatchChan chan readMicroBatch) {
 		for rows.Next() {
 			var pk, payload string
 			err = rows.Scan(&pk, &payload)
+			panicIfError(err)
 			rowsMap[pk] = payload
 		}
+		panicIfError(err)
+		rows.Close()
```
Comment (tzahij) [inline batchread/batchread.go:122]: For loop closes rows automatically (unless you break - which I dont)

File: batchread/batchread_test.go [hunk 2]
```diff
@@ -131,12 +129,9 @@ func collectStats(averageDurationCalculator *averageDurationType, duration time.
 	averageReadDurationStr := fmt.Sprintf("  %v  ", averageDurationCalculator.getAverage())
 	maxDurationStr := fmt.Sprintf("  %v  ", averageDurationCalculator.getMaxDuration())
 	minDurationStr := fmt.Sprintf("  %v  ", averageDurationCalculator.getMinDuration())
-
 	line := []string{runTypeStr, batchSizeStr, durationStr, averageReadDurationStr, maxDurationStr, minDurationStr, readNumStr}
 	err = w.Write(line)
 	panicIfError(err)
-	w.Flush()
```
Comment (tzahij) [inline batchread/batchread_test.go:138]: If you do not flush the CSV, good chances it will not be written

**Review by tzahij — CHANGES_REQUESTED** [review]

File: batchread/batchread_test.go (hunk 2 above)
Comment (nopcoder) [reply batchread/batchread_test.go:138]: line 117 - defer flush

File: batchread/batchread.go (hunk 1 above)
Comment (nopcoder) [reply batchread/batchread.go:122]: True, not use to panic I guess - because we should have handled the case of error which will not enter the loop.

Comment (nopcoder) [reply batchread/batchread.go]: btw, it means there is no need to switch case where we set 'err' in the reply - we only set not found - all other errors we panic.

Comment (tzahij) [reply batchread/batchread.go]: Right, but I think it should stay to show it is possible to pass other errors to he reader

File: batchread/batchread_test.go (hunk 2 above)
Comment (tzahij) [reply batchread/batchread_test.go:138]: What is the order of defer execution? the flush has to happen before the close

Comment (tzahij) [reply batchread/batchread_test.go:138]: Found out it is LIFO - no problem

**Review by tzahij — APPROVED** [review]

File: batchread/batchread.go (hunk 1 above)
Comment (nopcoder) [reply batchread/batchread.go]: so, keep the call to close so it will not be an optional resource leak in case they do handle the errors.

**Pull request merged**